### PR TITLE
Dogfood dagger with mage

### PR DESCRIPTION
**Pull request by dolanor**
- feat: use the current version of dagger for code generation
- build: build dagger with mage

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -49,3 +50,43 @@ func (Lint) Codegen(ctx context.Context) error {
...
+
+		src, err := core.Host().Workdir().Read().ID(ctx)
+		if err != nil {
+			return err
+		}
+
+		builder = builder.WithMountedDirectory("/app", src).WithWorkdir("/app")
+
+		builder = builder.Exec(api.ContainerExecOpts{
+			Args: []string{"mkdir", "/app/build"},
```
Comment (aluzzardi) [inline magefile.go:71]: @vito is there a way to do a mkdir through the API?

Comment (aluzzardi) [reply magefile.go:71]: Maybe this could be merged with #3264 -- `withNewDirectory()` without providing a `DirectoryID` creates an empty dir?

Comment (dolanor) [reply magefile.go:71]: I'd rather merge this now, with a note on #3264 to update it with the new API.
It will give a real use case to try on.

Comment (vito) [reply magefile.go]: Oops, missed this question, yeah we'll need #3264 for that. I like either adding `withNewDirectory` or just making the `DirectoryID` optional in #3264.

**Review by samalba — APPROVED** [review]
I agree with @dolanor that we should merge this now and see improve later when we can create a directory via the API. We need more examples like this one in the repo.

**Comment by aluzzardi** [comment]
rebased and pushed to your branch, waiting for CI to pass

**Comment by aluzzardi** [comment]
/cc @sipsma 

This should use the build cache

Trying to merge as is to get started, however the trick of copying only `go.mod` and running `go mod download` is not working anymore because of the sub-module

File: magefile.go [hunk 2]
```diff
@@ -55,3 +56,74 @@ func (Lint) Codegen(ctx context.Context) error {
...
+	}
+
+	builder = builder.WithMountedDirectory("/app", src).WithWorkdir("/app")
+
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"mkdir", "/app/build"},
+	})
+
+	builder = builder.Exec(dagger.ContainerExecOpts{
+		Args: []string{"go", "build", "-o", "/app/build/dagger", "/app/cmd/dagger"},
```
Comment (sipsma) [inline magefile.go:112]: We should set `CGO_ENABLED=0`, otherwise we'll probably be linked against musl due to importing `net` (if not now in the future)

**Review by sipsma — APPROVED** [review]
Comment about CGO is worth doing now, also saw the new failure in testing the CLI, think we just need to export the binary we build to somewhere in `PATH`. Otherwise shipit!

**Comment by aluzzardi** [comment]
@sipsma addressed!

**Pull request merged**